app/models/event_stream.rb

```diff
@@ -59,11 +61,19 @@ def self.event_groups
 
   def self.group_and_level(event_type)
     level = :detail # the level is detail as default
-    group, _ = event_groups.find do |_k, value|
-      GROUP_LEVELS.detect { |lvl| value[lvl]&.include?(event_type) }.tap do |level_found|
-        level = level_found if level_found
-      end
-    end
+    egroups = event_groups
+
+    group = egroups.find do |_, value|
+      GROUP_LEVELS
+        .detect { |lvl, _| value[lvl]&.include?(event_type) }
```

Comment:
Here I'd go with something like 

```ruby
.detect { |lvl| value[lvl]&.include?(event_type) || Regexp.new(value["#{lvl}_regex"]).match(event_type) }
```

and then we could omit all the `egroups` memoization, annonymous args `_` and the `.first`.

Reply:
This would break backwards compatibility, since badly formed regular expressions in one group of event types could take precedence over explicitly named events in other groups. And backwards compatibility is not something I would like to mess with.

Just to give you an example of bad situation, let us assume that one of the providers has this in its settings:

    :ems:
      :ems_dummy:
        :event_handling:
          :event_groups:
            :addition:
              :critical_regex:
              - .+

If we are unlucky and this is the first category that is checked, all of the events will be mapped to `:addition` group with `:critical` level. This example is an exaggeration, but the the general principle still stands.

We have two loops here for a reason: we would like to give explicitly named types higher precedence over regular expressions.